drivers/mpsl/clock_control/CMakeLists.txt

```diff
@@ -5,4 +5,7 @@
 #
 
 zephyr_library()
-zephyr_library_sources(clock_control_mpsl.c)
+zephyr_library_sources(nrfx_clock_mpsl.c)
+
+target_include_directories(zephyr_interface BEFORE INTERFACE
```

Comment:
It seems unnecessary to add this folder as an include folder

Reply:
that's the trick i mentioned. I do not set `CONFIG_NRFX_CLOCK` to not fetch `nrfx_clock.c` but there is a header `nrfx_power_clock.h` which handles interrupt handler depending on `nrfx_power` and `nrfx_clock` being enabled. I had to replace nrfx version with custom one placed in that folder. It's @anangl area so maybe he will have some better proposal.

Reply:
This is something to use with extreme care.
Used once, then people will start copying, and suddenly every module expects to have their includes first.

Example of an area where this is can be accepted use is in (unit) test-code, where a certain path should come first to avoid the need for specific test headers/code in production code.

Let's keep for now, and stabilize upmerge, and then i'll look at cleaning up.